apps/standalone/deploy/helm/flightctl-ui/values.yaml

```
@@ -1,23 +1,26 @@
-flightctlUi:
-  namespace: flightctl
-  image: quay.io/flightctl/flightctl-ui:latest
-  flightctlServer: https://flightctl-api:3443
-  flightctlMetricsServer: https://flightctl-api:9090
-  flightctlGrpcServer: flightctl-api-agent-grpc:7444
-  bootcImgUrl: "" #Bootc device image. Example: quay.io/example/example-agent-centos:bootstrap
-  qcow2ImgUrl: "" #Qcow2 device image. Example: https://example.com/disk.qcow2
-  internalOidcUrl: ""
-  baseUiUrl: ""
-  oidcInsecure: false
-  apiInsecure: false
-  certs:
-    ca: |
-      -----BEGIN
-      -----END
-    frontRouter: |
-      -----BEGIN
-      -----END
-    frontRouterKey: |
-      -----BEGIN
-      -----END
+## @param image.image Image to deploy.
+## @param image.tag Image tag to pull.
+## @param image.pullPolicy K8s pull policy for pulling the image.
+image:
+  image: quay.io/flightctl/flightctl-ui
+  tag: ""
+  pullPolicy: Always
+## @param baseURL Base URL of UI.
+baseURL: ""
+## @param auth.internalOidcAuthority URL of OIDC authority that is accessible by UI backend.
+## @param auth.insecureSkipTlsVerify Set to true if OIDC TLS certificate validation should be skipped.
+## @param auth.caCert A custom CA cert for OIDC TLS.
+auth:
+  internalOidcAuthority: ""
+  insecureSkipTlsVerify: false
+  caCert: ""
```

```suggestion
auth:
  internalOidcAuthority: ""
  insecureSkipTlsVerify: false
  caCert: ""
```

We should converge to put the auth details as globals, because I see the API also uses those.

https://github.com/flightctl/flightctl/blob/main/deploy/helm/flightctl/values.yaml#L65

What about having all of them as global.flightctl.auth   ?

---

Yes, I expect that we will provide these through globals. See that in the template I am using 
```
ca_oidc.crt: {{ default (.Values.global).flightctl.auth.caCert .Values.auth.caCert | quote }}
```

but I also add these in separate UI auth object in order to allow specifying different auth config than the global one. 
As maybe globally we want to use some `internalOidcAuthority` but for UI we want a different one. Im not sure why, but I think we want this flexibility.  WDYT ?

---

That would work, we just need to document the globals then :)

The bad side of adding too many knobs like this (global vs local) is that we need to maintain them later, I was considering on removing a lot of the knobs on the backend may be, but had not decided. 

What do you think? is there a possibility that we may need different oidc from the UI compared to the backend?

---

Yeah, the bad side here is that we need to maintain these ..
Lets imagine a case where you deploy
 - FCLT API and Keycloak to namespace FOO
 - FCTL UI to namespace BAR
 
 then you could set `global.auth.internalOIDC` to `https://keycloak` and UI auth will need to be `https://keycloak.FOO`

or you could set global to `https://keycloak.FOO` and it will work for both  ...
(but for this we would need to enable deploying UI to a different NS than API)

I guess it mostly depends on what we agree :)

---

Yeah, good thinking,

It would make sense to agree to document that when a global.internalOidcAuthority is set, we must provide the full cluster path for the URL, including the namespace. Then it may work on all cases. That we should probably do it regardless of keeping the specific knobs.


Another question I have is if we could auto-calculate that knob value if keycloak is enabled in the deployment. May be that is just some further refinement we could think of.